libbeat/asset/registry.go

```diff
@@ -56,6 +58,25 @@ func GetFields(beat string) ([]byte, error) {
 	return fields, nil
 }
 
+// GetFieldsFor returns a byte array containing all fields for the given beat
+// and given names
+func GetFieldsFor(beat string, names []string) ([]byte, error) {
+	var fields []byte
+	for _, name := range names {
+		data, ok := FieldsRegistry[beat][name]
```

Comment:
An idea I just had: What if we can allow to define dependencies when registering fields. For example the `system` fields depend on the `common` fields. And common fields depend on the `ecs` fields. Like this when generating the fields we could loop through the dependencies and get always the full result.

Reply:
I'll look into that when implementing it, but generally not planning on making any dependencies configurable.

Reply:
No need to look into it now. I think as you mentioned, outside of the scope of this PR.